### PR TITLE
Pretty print increment operators

### PR DESCRIPTION
**Pull request by atjn**
This is a bit of a special case because we represent increment operators as normal assignments. We have to add a bit of metadata to allow the pretty printer to work correctly.

### CODE REVIEW

File: src/main/java/dk/aau/p4/abaaja/Lib/Visitors/AstVisitor.java [hunk 1]
```diff
@@ -355,10 +355,12 @@ private void addProblem(ParserRuleContext ctx, String message) {
             addExpNode.set_lineNumber(ctx.start.getLine());
             addExpNode.set_lineEndNumber(ctx.getStop().getLine());
             if (ctx.op.getType() == mctlParser.INCREMENT) {
+                assStateNode.set_literalIncrement("++");
```
Comment (BaBrixx) [inline src/main/java/dk/aau/p4/abaaja/Lib/Visitors/AstVisitor.java:358]:
```suggestion
                assStateNode.set_postfixIncrement(true);
```

File: src/main/java/dk/aau/p4/abaaja/Lib/Visitors/AstVisitor.java [hunk 2]
```diff
@@ -355,10 +355,12 @@ private void addProblem(ParserRuleContext ctx, String message) {
...
                 addExpNode.set_operator(mctlParser.PLUS);
                 addExpNode.set_operatorLiteral("+");
             }
             else if (ctx.op.getType() == mctlParser.DECREMENT) {
+                assStateNode.set_literalIncrement("--");
```
Comment (BaBrixx) [inline src/main/java/dk/aau/p4/abaaja/Lib/Visitors/AstVisitor.java:363]:
```suggestion
                assStateNode.set_postfixIncrement(false);
```

File: src/main/java/dk/aau/p4/abaaja/Lib/Visitors/PrettyPrintVisitor.java [hunk 3]
```diff
@@ -205,8 +205,12 @@ public void visit(AssStateNode node){
         beginLineNode(node);
         printIndented();
         node.get_assignId().accept(this);
-        print(" = ");
-        node.get_assignExp().accept(this);
+        if(node.get_literalIncrement() != null){
+            print(node.get_literalIncrement());
```
Comment (BaBrixx) [inline src/main/java/dk/aau/p4/abaaja/Lib/Visitors/PrettyPrintVisitor.java:209]:
```suggestion
            print(node.get_postfixIncrement() ? "++" : "--");
```

File: src/main/java/dk/aau/p4/abaaja/Lib/Visitors/PrettyPrintVisitor.java [hunk 4]
```diff
@@ -205,8 +205,12 @@ public void visit(AssStateNode node){
         beginLineNode(node);
         printIndented();
         node.get_assignId().accept(this);
-        print(" = ");
-        node.get_assignExp().accept(this);
+        if(node.get_literalIncrement() != null){
```
Comment (BaBrixx) [inline src/main/java/dk/aau/p4/abaaja/Lib/Visitors/PrettyPrintVisitor.java:208]:
```suggestion
        if(node.get_postfixIncrement() != null){
```

**Review by BaBrixx — CHANGES_REQUESTED** [review]
Potentially change the implementation to the suggested one or change variable name to be _postfixOperatorLiteral

**Comment by atjn** [comment]
@BaBrixx is this wroom wroom?

**Review by BaBrixx — APPROVED** [review]
I like it, ......
Pablo_picasso_1

**Pull request merged**